src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,21 @@
+package duke;
+
+class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
```

Comment:
Perhaps you could rename "by" to something more descriptive.

```suggestion
    public Deadline(String description, String deadlineDate) {
        super(description);
        this.deadlineDate = deadlineDate;
    }
```